src/exchanges/populate.ts

```diff
@@ -0,0 +1,331 @@
+import { pipe, tap, map } from 'wonka';
+import { Exchange, Operation } from '../types';
+import {
+  DocumentNode,
+  buildClientSchema,
+  visitWithTypeInfo,
+  TypeInfo,
+  FragmentDefinitionNode,
+  SelectionSetNode,
+  GraphQLSchema,
+  IntrospectionQuery,
+  FragmentSpreadNode,
+  GraphQLObjectType,
+  GraphQLInterfaceType,
+  ASTNode,
+} from 'graphql';
+import { visit } from 'graphql';
+
+interface ExchangeArgs {
+  schema: IntrospectionQuery;
+}
+
+/** An exchange for auto-populating mutations with a required response body. */
+export const populateExchange = ({
+  schema: ogSchema,
+}: ExchangeArgs): Exchange => ({ forward }) => {
+  const schema = buildClientSchema(ogSchema);
+  /** List of operation keys that have already been parsed. */
+  const parsedOperations = new Set<number>();
+  /** List of operation keys that have not been torn down. */
+  const activeOperations = new Set<number>();
+  /** Collection of fragments used by the user. */
+  let userFragments: UserFragmentMap = {};
```

Comment:
`userFragments` and `typeFragments` should likely use `Object.create(null)` so we can ensure that weird fragment names don't mess things up (like `fragment prototype` or `fragment hasOwnProperty` 🙈 )

Reply:
Just checked this out. If we're going to go this route for security reasons we would need to handle this on a project wide scope.

This will be an issue on every use of spreads. I'll see if there are any babel/rollup plugins about which cover this.

![Screenshot 2019-11-20 at 15 30 32](https://user-images.githubusercontent.com/10779424/69252504-fe7a5f80-0baa-11ea-9d02-5bfe0f9045e6.png)

Reply:
We would have to take care of this every time we use an object as a record, yea, but (as far as I'm aware) we've taken care of that everywhere (?)
We rarely use objects as records anyway in `urql` (there are a lot of cases in Graphcache that do use `Object.create(null)`) but often we've also just gone for `Map` wherever possible

Reply:
`Object.create(null)` will do the trick - transforms prevent that spreading issue :+1: